microk8s-resources/default-hooks/remove.d/10-cni-link

```diff
@@ -4,11 +4,20 @@
 
 if ! is_strict || (is_strict && snapctl is-connected network-control)
 then
-  for link in cni0 calico.vxlan
-  do
-    if "${SNAP}/sbin/ip" link show "${link}"
-    then
-      "${SNAP}/sbin/ip" link delete "${link}" || true
-    fi
-  done
+	for link in cni0 vxlan.calico
+	do
+		if "${SNAP}/sbin/ip" link show "${link}"
+		then
+			"${SNAP}/sbin/ip" link delete "${link}" || true
+		fi
+	done
 fi
+
+echo "Gathering workload endpoints from $SNAP_DATA/.workload_endpoints"
+for workloadendpoint in $(cat "${SNAP_DATA}"/.workload_endpoints)
+do
+	if "${SNAP}/sbin/ip" link show "${workloadendpoint}"
+	then
+		"${SNAP}/sbin/ip" link delete "${workloadendpoint}" || true
+	fi
+done
```

Comment:
nit; can you preserve the 2 spaces for indentation?

Reply:
Hey @neoaggelos , thanks a lot for the feedback and the pointers on testing!

_Note I have preserved the 2 spaces and pushed the change, sorry about that :)_

For testing, I tested the change locally (see Testing paragraph in the PR description), however it makes sense to have it in CI. I think this could be an approach:
Reuse the cluster used in the link you shared, there is already an option to install a local snap (via setting the channel env variable), and I would

* build the charm locally (needs `charmcraft pack`)

* install it (is covered by the `_transfer_install_local_snap_lxc` function)

* place a check before teardown of the VMs to uninstall microk8s and check that there are no interfaces left (on all machines part of the cluster) (needs an additional function)

What do you think?

_I want to reuse as much of the existing testing code as possible, though I can create an additional separate test file with independent setup/teardown.._

Reply:
Hey @neoaggelos , I added a test for this in `tests/test-cluster.py` - I wanted to make use of the VM class - the test automates my manual test: spin up a VM, install microk8s, wait for microk8s to be ready and calico to be up and running, remove the microk8s snap and ensure there are no calico interfaces present.